packages/template-retail-react-app/app/pages/product-detail/index.test.js

```diff
@@ -904,6 +904,145 @@ test('Add to Cart (Pick Up in Store) includes inventoryId for the selected varia
+    const variantWithInventoryId = '701642811398M'
+    const variantWithoutInventoryId = '701642841227M'
+
+    let currentProductData = null
+
+    // Mock API that returns different products based on the pid parameter
+    global.server.use(
+        rest.get('*/products/:productId', (req, res, ctx) => {
+            const url = new URL(req.url)
+            const pidParam = url.searchParams.get('pid')
+            const requestedId = pidParam || req.params.productId
+
+            if (requestedId === variantWithInventoryId) {
+                currentProductData = {
+                    ...masterProduct,
+                    id: variantWithInventoryId,
+                    name: 'Long Sleeve Crew Neck - Fire Red',
+                    inventories: [
+                        {
+                            id: inventoryId,
+                            orderable: true,
+                            ats: 5,
+                            stockLevel: 5
+                        }
+                    ]
+                }
+            } else if (requestedId === variantWithoutInventoryId) {
+                currentProductData = {
+                    ...masterProduct,
+                    id: variantWithoutInventoryId,
+                    name: 'Long Sleeve Crew Neck - Out of Stock',
+                    inventories: [
+                        {
+                            id: inventoryId,
+                            orderable: false,
+                            ats: 0,
+                            stockLevel: 0
+                        }
+                    ]
+                }
+            } else {
+                currentProductData = {
+                    ...masterProduct,
+                    id: masterProductId,
+                    inventories: [
+                        {
+                            id: inventoryId,
+                            orderable: true,
+                            ats: 10,
+                            stockLevel: 10
+                        }
+                    ]
+                }
+            }
+
+            return res(ctx.json(currentProductData))
+        }),
+        rest.get('*/customers/:customerId/baskets', (req, res, ctx) => {
+            return res(ctx.json(mockCustomerBaskets))
+        })
+    )
+
+    // Start with master product
+    window.history.pushState({}, 'ProductDetail', `/uk/en-GB/product/${masterProductId}`)
+    renderWithProviders(<MockedComponent />)
+
+    // Wait for initial load
+    expect(await screen.findByTestId('product-details-page')).toBeInTheDocument()
+    await waitFor(() => {
+        expect(screen.getByRole('link', {name: /mens/i})).toBeInTheDocument()
+    })
+
+    // Select pickup
+    const pickupLabel = await screen.findByLabelText(/Pick Up in Store/i)
+    fireEvent.click(pickupLabel)
+    expect(pickupLabel).toBeChecked()
+
+    // Navigate to variant with inventory (simulate variant selection)
+    await act(async () => {
+        window.history.pushState(
```

Comment:
Instead doing this, can we minic what users are doing by clicking variation buttons? 🤔